### DISCUSSION TITLE
GUI for windows version of whisper.cpp

### DISCUSSION THREAD

**Discussion by Topping1** [ggml-org/whisper.cpp#189] [Show and tell]
@ggerganov not sure if this is the best place to post this but I have developed a simple GUI for the windows version of whisper.cpp (can be adapted easily for linux too). Project is in python using the Gooey library. In the repository [https://github.com/Topping1/whispercppGUI](https://github.com/Topping1/whispercppGUI) you can find the source code and windows binary (x64).

-Update 1 (16-Jan-2023): whispercppGUI can now use FFMPEG to open files directly, without manual conversion to a WAV file, can use a variable speedup option to the audio file, includes a textbox to add custom parameters to the command line.

**Comment by ggerganov** [comment]
Thanks for sharing!
I'll move this to a "Show and tell" discussion.
Maybe you can add a screenshot of the GUI for those of us who don't have Windows.

**Comment by Topping1** [comment]
whisper.cppGUI is a simple GUI for the Windows x64 binary of whisper.cpp. The easiest way to get the most updated windows binary is to download them from the [actions](https://github.com/ggerganov/whisper.cpp/actions) page of the whisper.cpp project. For example select the first item on the workflows list shown in the figure below, 

![image](https://user-images.githubusercontent.com/78745143/204151097-02ddc17e-c4b7-4d10-af33-4ad266f5c2fe.png)

then scroll down to find the "Artifacts produced during runtime" and download the "whisper-bin-x64.zip" file.
![image](https://user-images.githubusercontent.com/78745143/204151179-b1f81476-f9be-4734-9e61-eef1a060bcd9.png)

you can get the binary for whisper.cppGUI from this [repository](https://github.com/Topping1/whispercppGUI).
Unzip the files in the same folder and execute whisperGUI.exe (please note that you also need to download a .bin model to start the transcribing process). The main window looks like this:

![screenshot01](https://user-images.githubusercontent.com/78745143/204151446-8c7c3b7b-4ab5-42d1-a0f9-ac17c2f8112a.png)

for now, only a subset of options are included. Please let me know if you want me to include more options.

The program also includes a link to the .bin models download page on Huggingface:
![screenshot02](https://user-images.githubusercontent.com/78745143/204151530-e4cf8169-92c9-40f6-a811-d2f1c37ed3ba.png)
![screenshot03](https://user-images.githubusercontent.com/78745143/204151542-5f450ad7-287e-466d-bf42-8fa628769e1f.png)

the following figures show a simple run of the program:
![screenshot04](https://user-images.githubusercontent.com/78745143/204151578-1707a714-bca2-4459-8318-3718e84f5955.png)

![screenshot05](https://user-images.githubusercontent.com/78745143/204151579-87613eb7-b8cb-4f69-91af-d357a0875ba6.png)

I hope you all find this GUI useful.

**Reply by dgoryeo** [reply]
Great work --and thanx for the installatuon guide. 

Quick suggestion / feature recommendation: I think it would be good to have a GUI option to pass any additional arguments to Whisper call. What I mean is a an input box where the user can write/input any additional Whisper arguments that the user wants to pass through, example --beam = 5. 

I am just running my first try and can see that there are halucination text --which usually happens with large model if there is a long pause between speech sections. I think it would be great if one could combine a VAD with the GUI. That helps a lot with non-English languages.

**Reply by Topping1** [reply]
@dgoryeo Great suggestion! I will look into including an input box for extra commands.

**Reply by Topping1** [reply]
> Great work --and thanx for the installatuon guide.
> 
> Quick suggestion / feature recommendation: I think it would be good to have a GUI option to pass any additional arguments to Whisper call. What I mean is a an input box where the user can write/input any additional Whisper arguments that the user wants to pass through, example --beam = 5.
> 
> I am just running my first try and can see that there are halucination text --which usually happens with large model if there is a long pause between speech sections. I think it would be great if one could combine a VAD with the GUI. That helps a lot with non-English languages.

A new version of whispercppGUI is out with ffmpeg support. Additional arguments are next on the list.

**Reply by Topping1** [reply]
Additional arguments are now supported

**Comment by rosmur** [comment]
This is really great stuff! Unfortunately Im on mac almost exclusively now. Could you possibly make a mac version? I unfortunately dont have the skills to do that myself

**Reply by AlexPeregudov** [reply]
It's not a problem
If You have Python installed
1. Install the Python dependencies as described
`pip install gooey`
2. Get [whisperGUI.py](https://github.com/Topping1/whispercppGUI/blob/main/whisperGUI.py)
3. Open it on text editor
4. Replace 'main.exe' to  'main' in target string & save.
5. Copy whisperGUI.py to local whisper.cpp folder.
6. Switch to bright macOS theme (dark theme not fully supported).
7. Run
`pyhton whisperGUI.py` in Terminal.

May be, author of whisperGUI.py can add macOS detection and replacing main.exe to main.

**Comment by aaaaaammmmmm** [comment]
Thanks, this is very useful for people not used to working in a CLI. 
One thing that would make it even more useful is if you could run the ffmpeg automatically before transcribing. This would make it compatible with virtually all video and audio files instead of only the format supported by Whisper.cpp. No files are this format to begin with, so all files has to be converted with ffmpeg or other software which makes it hard for novice users. 
I believe the normal Whisper supports all files supported by ffmpeg.

**Reply by Topping1** [reply]
> Thanks, this is very useful for people not used to working in a CLI. One thing that would make it even more useful is if you could run the ffmpeg automatically before transcribing. This would make it compatible with virtually all video and audio files instead of only the format supported by Whisper.cpp. No files are this format to begin with, so all files has to be converted with ffmpeg or other software which makes it hard for novice users. I believe the normal Whisper supports all files supported by ffmpeg.

@aaaaaammmmmm  Unfortunately I was low on free time last month or so and could not update the GUI script. I have an update in the works that can use ffmpeg (not automatically, I haven't found a way to do it with the Gooey library...yet) and also has a simple GUI for youtube-dlp.

**Reply by Topping1** [reply]
> Thanks, this is very useful for people not used to working in a CLI. One thing that would make it even more useful is if you could run the ffmpeg automatically before transcribing. This would make it compatible with virtually all video and audio files instead of only the format supported by Whisper.cpp. No files are this format to begin with, so all files has to be converted with ffmpeg or other software which makes it hard for novice users. I believe the normal Whisper supports all files supported by ffmpeg.

@aaaaaammmmmm the new version of whispercppGUI is out. Now it can use ffmpeg to automatically convert the input file before running whispercpp.
